pkg/services/alerting/rule.go

```diff
@@ -137,7 +138,8 @@ func NewRuleFromDBAlert(ruleDef *models.Alert) (*Rule, error) {
 	for _, v := range ruleDef.Settings.Get("notifications").MustArray() {
 		jsonModel := simplejson.NewFromAny(v)
 		if id, err := jsonModel.Get("id").Int64(); err == nil {
-			model.Notifications = append(model.Notifications, fmt.Sprintf("%09d", id))
+			maybeUID := translateNotificationIDToUID(id, ruleDef.OrgId)
```

Comment:
As I suggested in another comment I think that `translateNotificationIDToUID` should return also the error
which it should be checked here and if it's not `null` it should return a `ValidationError` similar to this [one](https://github.com/grafana/grafana/blob/6cd001f09a7c1e7ed1fc10a8302c9fcfed9a0d93/pkg/services/alerting/rule.go#L146).

Reply:
agreed